src/core/qgsgdalutils.cpp

```diff
@@ -490,6 +490,18 @@ GDALDatasetH QgsGdalUtils::rpcAwareAutoCreateWarpedVrt(
   return GDALAutoCreateWarpedVRTEx( hSrcDS, pszSrcWKT, pszDstWKT, eResampleAlg, dfMaxError, psOptionsIn, opts );
 }
 
+void *QgsGdalUtils::rpcAwareCreateTransformer( GDALDatasetH hSrcDS, GDALDatasetH hDstDS, char **papszOptions )
+{
+  if ( GDALGetMetadata( hSrcDS, "RPC" ) )
+  {
+    // well-behaved RPC should have height offset a good value for RPC_HEIGHT
+    const char *heightOffStr = GDALGetMetadataItem( hSrcDS, "HEIGHT_OFF", "RPC" );
+    if ( heightOffStr )
+      papszOptions = CSLAddNameValue( papszOptions, "RPC_HEIGHT", heightOffStr );
```

Comment:
you should CSLDuplicate() the input papszOptions, and CSLDestroy() it after use. Currently, the caller of rpcAwareCreateTransformer() might CSLDestroy() something that has been destroyed by CSLAddNameValue()

Reply:
Ah good point, thanks - fixed in 75965e5